File: frontend/src/Book.ts
export const ReadingStatusEnum = {
  NotStarted: 0,
  Reading: 1,
  Finished: 2,
  Abandoned: 3
} as const;

export type ReadingStatus = typeof ReadingStatusEnum[keyof typeof ReadingStatusEnum];

export interface Book {
  id?: number
  title: string
  author: string
  genre: string
  notes?: string
  rating?: number
  status: ReadingStatus
}

File: BookTracker/Models/Book.cs
namespace BookTracker.Models;

public class Book
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Author { get; set; } = string.Empty;
    public string Genre { get; set; } = string.Empty;
    public string? Notes { get; set; }
    public int? Rating { get; set; } // 1-5 stars

    public ReadingStatus Status { get; set; } = ReadingStatus.NotStarted;
}
